### PR TITLE
Minor changes refactor

### PR DESCRIPTION
**Pull request by Salisbum**

### CODE REVIEW

File: spec/features/user_adds_a_new_landmark_spec.rb [hunk 1]
```diff
@@ -6,23 +6,27 @@
...
 
     click_button "Submit Landmark"
 
     expect(page).to have_content "Landmark added successfully!"
-    expect(page).to have_content "Statue of Liberty"
-    expect(page).to have_content "New York"
-    page.should have_selector ("img[src$='https://upload.wikimedia.org/wikipedia/commons/thumb/a/a1/Statue_of_Liberty_7.jpg/500px-Statue_of_Liberty_7.jpg']")
-    expect(page).to have_content "A colossal neoclassical sculpture on Liberty Island"
+    expect(page).to have_content landmark.name
+    expect(page).to have_content landmark.location
```
Comment (dreamingfurther) [inline spec/features/user_adds_a_new_landmark_spec.rb]: So. I have a real issue with using variables in assertions. The issue is, what if for some reason `landmark.name` returned `nil`? 

Now your test would be asserting `expect(page).to have_content(nil)`, which will _always_ return true. At least in `expect` assertions I think it's really important to have hard coded strings or other values.

File: spec/features/user_edits_an_existing_landmark_spec.rb [hunk 2]
```diff
@@ -1,43 +1,30 @@
...
+    fill_in 'Description', with: landmark.description
 
     click_button "Submit Landmark"
 
     expect(page).to have_content "Landmark updated successfully!"
-    expect(page).to have_content "Statue of Liberty"
-    expect(page).to have_content "New York"
-    page.should have_selector ("img[src$='https://upload.wikimedia.org/wikipedia/commons/thumb/a/a1/Statue_of_Liberty_7.jpg/500px-Statue_of_Liberty_7.jpg']")
-    expect(page).to have_content "A colossal neoclassical sculpture on Liberty Island"
+    expect(page).to have_content landmark.name
```
Comment (dreamingfurther) [inline spec/features/user_edits_an_existing_landmark_spec.rb]: same comment as above on variable calls in test assertions.

File: spec/features/user_adds_a_new_landmark_spec.rb (hunk 1 above)
Comment (Salisbum) [inline spec/features/user_adds_a_new_landmark_spec.rb]: @dreamingfurther Interesting, cause that conflicts with feedback I'd gotten previously where the reviewer preferred variables over hardcoded strings given how you might update your factories.

File: app/models/landmark.rb [hunk 3]
```diff
@@ -6,6 +6,8 @@ class Landmark < ActiveRecord::Base
   validates :location, presence: true
   validates :image, presence: true
   validates :description, presence: true
+  validates :user_id, presence: true
```
Comment (radavis) [inline app/models/landmark.rb]: I would prefer to see `validates :user, presence: true`. Ids are something the database takes care of. In Ruby land, we care about objects.

File: spec/features/user_creates_review_spec.rb [hunk 4]
```diff
@@ -49,7 +51,8 @@
       name: "Statue of Liberty",
       location: "New York",
       image: "https://upload.wikimedia.org/wikipedia/commons/thumb/a/a1/Statue_of_Liberty_7.jpg/500px-Statue_of_Liberty_7.jpg",
-      description: "A colossal neoclassical sculpture on Liberty Island"
+      description: "A colossal neoclassical sculpture on Liberty Island",
+      user: FactoryGirl.create(:user)
```
Comment (radavis) [inline spec/features/user_creates_review_spec.rb]: You can move this task of user creation to your factory.

File: db/seeds.rb [hunk 5]
```diff
@@ -1 +1,184 @@
...
+  location: "Paris, FR",
+  image: "http://cdn.history.com/sites/2/2015/04/hith-eiffel-tower-iStock_000016468972Large.jpg",
+  description: "Very tall, lots of metal.",
+  user: user1
+)
+
+landmark2 = Landmark.create(
+  name: "Grand Canyon",
+  location: "USA",
+  image: "https://media.deseretdigital.com/file/b41fa91284?crop=top:0%7Cleft:0%7Cwidth:1260%7Cheight:670%7Cgravity:Center&resize=width:1260&order=resize,crop&c=14&a=c856f78c",
```
Comment (Salisbum) [inline db/seeds.rb]: is a url

File: app/controllers/landmarks_controller.rb [hunk 6]
```diff
@@ -19,11 +19,12 @@ def new
 
   def create
     @landmark = Landmark.new(landmark_params)
+    @landmark.user = current_user
     if @landmark.save
       flash[:notice] = "Landmark added successfully!"
       redirect_to landmark_path(@landmark)
     else
-      flash[:error] = "Landmark not added. #{@landmark.errors.full_messages.join ', '}."
+      flash[:error] = "Landmark not added successfully! #{@landmark.errors.full_messages.join ', '}."
```
Comment (radavis) [inline app/controllers/landmarks_controller.rb]: Kind of a weird message.

**Pull request merged**